feat(event_list): change tooltip style

dqunbp opened the pull request:
change disaster card stats tooltip style
https://kontur.fibery.io/Tasks/Task/Change-tooltips-style-on-disaster-card-14451

dqunbp committed:
feat(event_list): change tooltip style

change disaster card stats tooltip style

IlyaIzr commented:
1. Task says "Add tooltip «Open timeline» for play button" - i don't see that on the code
2. Branch doesn't run locally 
![image](https://user-images.githubusercontent.com/50058726/219315945-81d0aa5f-3552-4a8c-ae66-509800c3d989.png)

Code that is presented so far looks good

propakov commented:
@IlyaIzr on my pc runs ok locally

dqunbp committed:
feat(event_list): add tooltip to timeline button

add tooltip to timeline button
move translatons for severity from common to event_list key

propakov commented on src/features/events_list/components/EpisodeTimelineToggle/EpisodeTimelineToggle.tsx:
```diff
@@ -1,17 +1,31 @@
+        {isActive && panelState.isOpen ? <PlayActive24 /> : <Play24 />}
+      </span>
+      <Tooltip open={showTooltip} hoverBehavior>
+        <Trans i18nKey="event_list.open_timeline_button">Open timeline</Trans>
```
hm, `<Trans />` should only be used if there is html content inside translation string
here it seems that you need just {i18n.t("event_list.open_timeline_button")}

propakov commented:
Have lots of warnings in console when just open the map without any actions
And timeline's 'Play button' tooltip not working
<img width="1536" alt="Screenshot 2023-02-17 at 09 25 22" src="https://user-images.githubusercontent.com/20676525/219565857-2ff5c517-9721-4ee3-acbe-748ea6e87b52.png">

propakov commented:
could you also please make SEVERITY_TO_TEXT as a function that returns translation, or put this const inside the component? 
i18.n should be called from component, not just on app start
now it's a bug (try to show tooltip after lng switch): 
<img width="330" alt="Screenshot 2023-02-17 at 09 42 55" src="https://user-images.githubusercontent.com/20676525/219569194-d29b3da6-321c-4bc3-be2e-9968e939c894.png">

dqunbp committed:
Update translate funtions

Move severity translations to fn
Update analytycs translation fn

dqunbp committed:
Fix timeline button tooltip

dqunbp commented:
> could you also please make SEVERITY_TO_TEXT as a function that returns translation, or put this const inside the component? i18.n should be called from component, not just on app start now it's a bug (try to show tooltip after lng switch): <img alt="Screenshot 2023-02-17 at 09 42 55" width="330" src="https://user-images.githubusercontent.com/20676525/219569194-d29b3da6-321c-4bc3-be2e-9968e939c894.png">

Now we don't have a translation for this

dqunbp commented:
> Have lots of warnings in console when just open the map without any actions And timeline's 'Play button' tooltip not working <img alt="Screenshot 2023-02-17 at 09 25 22" width="1536" src="https://user-images.githubusercontent.com/20676525/219565857-2ff5c517-9721-4ee3-acbe-748ea6e87b52.png">

Fixed, `targetRef` prop was skipped

propakov commented:
> > could you also please make SEVERITY_TO_TEXT as a function that returns translation, or put this const inside the component? i18.n should be called from component, not just on app start now it's a bug (try to show tooltip after lng switch): <img alt="Screenshot 2023-02-17 at 09 42 55" width="330" src="https://user-images.githubusercontent.com/20676525/219569194-d29b3da6-321c-4bc3-be2e-9968e939c894.png">
> 
> Now we don't have a translation for this

yeah, but it will work after translation appear, so no problem here 👍

propakov commented on src/components/SeverityIndicator/SeverityIndicator.tsx:
```diff
@@ -2,23 +2,24 @@ import { Tooltip } from '@konturio/ui-kit';
+  };
+
+  return translations[severity];
 };
```
just not to call all translations even if they not needed: 
```suggestion
const severityToText = (severity: Severity) => {
  switch (severity) {
    case 'UNKNOWN':
      return i18n.t('event_list.severity_unknown');
    case 'TERMINATION':
      return i18n.t('event_list.severity_termination');
    case 'MINOR':
      return i18n.t('event_list.severity_minor');
    case 'MODERATE':
      return i18n.t('event_list.severity_moderate');
    case 'SEVERE':
      return i18n.t('event_list.severity_severe');
    case 'EXTREME':
      return i18n.t('event_list.severity_extreme');
  }
};
```

dqunbp replied:
I thought about switch-case but decided that the object be a more compact construction

propakov reviewed (approved):
Works good 👍 
Found one minor improvement

dqunbp committed:
Update src/components/SeverityIndicator/SeverityIndicator.tsx

Co-authored-by: Pavel Ropakov <propakov@kontur.io>
